project_generator/tools/uvision.py

```diff
@@ -215,6 +215,20 @@ def _uvproj_set_TargetArmAds(self, uvproj_dic, project_dic):
         uvproj_dic['Cads']['VariousControls']['Define'] = ', '.join(project_dic['macros']).encode('utf-8')
         uvproj_dic['Aads']['VariousControls']['Define'] = ', '.join(project_dic['macros']).encode('utf-8')
 
+        pro_def = ProGenDef(tool_name)
+        if not pro_def.is_supported(project_dic['target'].lower()):
+            raise RuntimeError("Target %s is not supported. Please add them to https://github.com/project-generator/project_generator_definitions" % project_dic['target'].lower())
+        mcu_def_dic = pro_def.get_tool_definition(project_dic['target'].lower())
+        if not mcu_def_dic:
+             raise RuntimeError(
+                "Target definitions were not found for %s. Please add them to https://github.com/project-generator/project_generator_definitions" % project_dic['target'].lower())
+        try:
+            uvproj_dic['ArmAdsMisc']['useUlib'] = mcu_def_dic['TargetOption']['uLib'][0]
+            uvproj_dic['ArmAdsMisc']['OptFeed'] = mcu_def_dic['TargetOption']['OptFeed'][0]
```

Comment:
``OptFeed`` what does it do? I commented on progen def, I would leave ``uLib`` out of definitions